examples/inspection/plot_partial_dependence.py

```diff
@@ -163,179 +298,202 @@
+# features to see the difference:
 print("Computing partial dependence plots...")
+features_info = {
+    "features": ["temp", "humidity", ("temp", "humidity")],
+    "kind": "average",
+}
+_, ax = plt.subplots(ncols=3, figsize=(10, 4))
 tic = time()
 display = PartialDependenceDisplay.from_estimator(
-    est_no_interactions,
+    hgbdt_model_without_interactions,
     X_train,
-    ["MedInc", "AveOccup", "HouseAge", "AveRooms"],
-    kind="both",
-    subsample=50,
-    n_jobs=3,
-    grid_resolution=20,
-    random_state=0,
-    ice_lines_kw={"color": "tab:blue", "alpha": 0.2, "linewidth": 0.5},
-    pd_line_kw={"color": "tab:orange", "linestyle": "--"},
+    **features_info,
+    ax=ax,
+    **common_params,
 )
-
 print(f"done in {time() - tic:.3f}s")
-display.figure_.suptitle(
-    "Partial dependence of house value with Gradient Boosting\n"
-    "and no interactions allowed"
+_ = display.figure_.suptitle(
+    "1-way vs 2-way of numerical PDP using gradient boosting", fontsize=16
 )
-display.figure_.subplots_adjust(wspace=0.4, hspace=0.3)
+plt.subplots_adjust(wspace=0.3)
 
 # %%
-# All 4 plots have parallel ICE lines meaning there is no interaction in the
-# model.
-# Let us also have a look at the corresponding 2D-plot.
-
+# Although the 2D-plot shows much less interaction compared with the 2D-plot from
+# above, it is much harder to come to the conclusion that there is no interaction at
+# all. This might be a cause of the discrete predictions of trees in combination with
+# numerically precision of partial dependence. We also observe that the univariate
+# dependence plots have slightly changed as the model tries to compensate for the
+# forbidden interactions (spikes in the "humidity" PD plot).
+#
+# The partial dependence between categorical features will provide a discrete
+# reprensentation that can be shown as a heatmap. For instance the interaction between
+# the season, the weather, and the target would be as follow:
 print("Computing partial dependence plots...")
+features_info = {
+    "features": ["season", "weather", ("season", "weather")],
+    "kind": "average",
+    "categorical_features": categorical_features,
+}
+_, ax = plt.subplots(ncols=3, figsize=(14, 4))
 tic = time()
-_, ax = plt.subplots(ncols=3, figsize=(9, 4))
 display = PartialDependenceDisplay.from_estimator(
-    est_no_interactions,
+    hgbdt_model,
     X_train,
-    ["AveOccup", "HouseAge", ("AveOccup", "HouseAge")],
-    kind="average",
-    n_jobs=3,
-    grid_resolution=20,
+    **features_info,
     ax=ax,
+    **common_params,
 )
+
 print(f"done in {time() - tic:.3f}s")
-display.figure_.suptitle(
-    "Partial dependence of house value with Gradient Boosting\n"
-    "and no interactions allowed"
+_ = display.figure_.suptitle(
+    "1-way vs 2-way PDP of categorical features using gradient boosting", fontsize=16
 )
-display.figure_.subplots_adjust(wspace=0.4, hspace=0.3)
+plt.subplots_adjust(wspace=0.3)
```

Comment:
This does not seem to work as expected: the x tick labels are truncated:

![image](https://user-images.githubusercontent.com/89061/203120001-ddbbf7de-8081-4fe7-a661-b5f6c75fb296.png)

I would be in favor of trying to use the constrained layout every time it's needed in this example instead of manual subplot adjustments:

- https://matplotlib.org/stable/tutorials/intermediate/constrainedlayout_guide.html

This might or might not impose an update of our minimum version requirement for the matplotlib dependency but I am fine with that if it can make the example more readable.

Reply:
I grepped our code and we already have examples that use `plt.subplots(constrained_layout=True)` instead of the newstyle parameter `plt.subplots(layout="constrained")`.

Let's try to see if `plt.subplots(constrained_layout=True)` works.